clang/lib/Sema/SemaOpenACC.cpp

```diff
@@ -2589,6 +2589,59 @@ SemaOpenACC::ActOnOpenACCAsteriskSizeExpr(SourceLocation AsteriskLoc) {
   return BuildOpenACCAsteriskSizeExpr(AsteriskLoc);
 }
 
+/// Loops through a type and generates an appropriate InitListExpr to generate
+/// type initialization.
+static Expr *GenerateReductionInitRecipeExpr(ASTContext &Context,
```

Comment:
Maybe this could have an enum argument indicating the required initializer kind (zero, one, least, largest, all 1's, etc.).

Reply:
Yeah, I was thinking a few things, either that, or a function object/function pointer that knows how to do the lowering.  I was intending to do that as a part of my patch though.